connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -115,7 +115,7 @@ public void initialize(TaskConfig taskConfig) {
     }
 
     protected void close() {
-        // nothing to do
+        producer.close();
```

Comment:
One question: would it make sense to use a timeout when closing the producer? Or would we have already flushed by this time?